### PR TITLE
Map `increments` to `integer` in non-primary fields

### PR DESCRIPTION
**Pull request by craicoverflow**
Fixes https://github.com/aerogear/graphback/issues/831

Non-primary columns were getting created with the SERIAL data type, which is intended as a primary key field. Issues arose when running a migration against the database

```
alter table "public"."note" alter column "ownerId" type serial primary key using ("ownerId"::serial primary key) - syntax error at or near "primary"
```

This fix maps `increments` (SERIAL) to `integer` for foreign key fields.

## Other

- Add filter to prevent table deletion in runtime template.
- enable `@oneToOne` and `@oneToMany` markers in `graphql-migrations`. migrations will use `key` as the preferred way to create a foreign key field.

### CODE REVIEW

File: packages/graphql-migrations/src/abstract/generateAbstractDatabase.ts [hunk 1]
```diff
@@ -232,7 +234,7 @@ class AbstractDatabaseBuilder {
 
       // Object
     } else if (isObjectType(fieldType)) {
-      columnName = annotations.name || `${field.name}Id`
+      columnName = relationshipMarker?.key || annotations.name || `${field.name}Id`
```
Comment (wtrocki) [inline packages/graphql-migrations/src/abstract/generateAbstractDatabase.ts]: That could be nightmare to document. Do we need both annotations and marker support at the time?

**Comment by wtrocki** [comment]
Do we have any tests that will simulate creation and later migration in this package? Trying to figure out how to protect from similar issues in near future

File: packages/graphql-migrations/src/abstract/generateAbstractDatabase.ts (hunk 1 above)
Comment (craicoverflow) [reply packages/graphql-migrations/src/abstract/generateAbstractDatabase.ts]: We discussed supporting the existing annotation but just documenting the new one in Graphback.

**Comment by craicoverflow** [comment]
> Do we have any tests that will simulate creation and later migration in this package? Trying to figure out how to protect from similar issues in near future

I don't believe we do - our tests use SQLite which works only for a single migration.

**Pull request merged**